packages/phone-number-privacy/signer/src/key-management/key-provider-base.ts

```diff
@@ -1,28 +1,51 @@
+export enum KeyType {
+  phoneNumberPrivacy = 'phoneNumberPrivacy',
+  domains = 'domains',
+}
+export interface Key {
+  type: KeyType
+  version: number
+}
+
+export const keys: Key[] = [
+  {
+    type: KeyType.phoneNumberPrivacy,
+    version: 1,
+  },
+  {
+    type: KeyType.phoneNumberPrivacy,
+    version: 2,
+  },
+  {
+    type: KeyType.domains,
+    version: 1,
+  },
+]
 export interface KeyProvider {
-  fetchPrivateKeyFromStore: () => Promise<void>
-  getPrivateKey: () => string
+  fetchPrivateKeyFromStore: (key: Key) => Promise<void>
+  getPrivateKey: (key: Key) => string
 }
 
 const PRIVATE_KEY_SIZE = 72
 
 export abstract class KeyProviderBase implements KeyProvider {
-  protected privateKey: string | null = null
+  protected privateKeys: Map<Key, string> = new Map()
 
-  public getPrivateKey() {
-    if (!this.privateKey) {
+  public getPrivateKey(key: Key) {
+    const privateKey = this.privateKeys.get(key)
+    if (!privateKey) {
       throw new Error('Private key is empty, provider not properly initialized')
```

Comment:
It's now the case that the user could have requested a key version that does not exist. Suggested change to the error message accounts for that.
```suggestion
      throw new Error(`Private key is unavailable: ${JSON.stringify(key)}`)
```